iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -900,7 +900,8 @@ private ushort GetNextPacketId()
             Debug.Assert(lockTaken);
             unchecked
             {
-                ret = ++_packetId == 0 ? ++_packetId : _packetId;
+                _packetId = (ushort)(++_packetId % 0x8000);
```

Comment:
Should better use `++packetId&0x7FFF` instead. `%` is much worse than `&` to permanence concern.

Reply:
for me, "Modulo" expresses more the intention that "the result should not be greater than" - as a clock arithmetic,  it is usual to use it to restrict numbers into a certain range and I think it is more readable because of this (I even considered using an "if" to make the code less cryptic :) ) But probably it is more like a matter of preference. As this is not a central peace of a loop calculating something, I would not worry about the performance hit.